### ISSUE TITLE
Remove autoconf files from distribution

### ISSUE THREAD

**Issue by torognes** [torognes/vsearch#139]
The following files should be included in the distribution:
aclocal.m4
config.h.in
configure
compile
depcomp
install-sh
missing
Makefile.in

This should allow compilation even when autoconf is not installed.

**Comment by xflouris** [comment]
I'm not sure if all these files are needed for the Git repository.
On a machine with autotools installed, the command

`make distcheck`

will create a tarball that does not require autotools. We can then include this in the release tag.

**Comment by xflouris** [comment]
> For me, the repository in its current form creates a perfectly working tarball.

Sorry, I take it back. It was possible to create a tarball in the previous version, in this one, it seems there is a problem finding cityhash/city.h

```
../../src/vsearch.h:101:27: fatal error: cityhash/city.h: No such file or directory
 #include "cityhash/city.h"
```

For now, `make dist` works, but `make distcheck` must pass as well.
I'll check what the problem is. I believe we're only missing the dependency of city.h in src/Makefile.am

**Comment by torognes** [comment]
Thanks for looking into this, @xflouris

**Comment by xflouris** [comment]
The simple fix to have `make distcheck` working again:

`cityhash/city.h` must be added to `libcpu_sse2_a_SOURCES`, `libcpu_ssse3_a_SOURCES`and `__top_builddir__bin_vsearch_SOURCES` in src/Makefile.am

This will also require that line

```
#include <city.h>
```
in src/cityhash/city.cc is changed to

```
#include "city.h"
```

**Comment by torognes** [comment]
Ok, I'll fix that in the next release.

Do you think we can skip some of the files listed above for the release? The files listed were those that I found were included in the tarball created from running `make dist`.

**Comment by xflouris** [comment]
Sorry for the late reply. I don't think we need to include any of the 8 files in the Git repository. They're automatically generated by autotools and may differ from version to version and Ttey're included in the distribution tarballs. Hence, if someone wants to only compile the rel

**Comment by torognes** [comment]
But the distribution tarballs on the release page on Github is created by Github automatically as a snapshot of the files in the repo. The tarballs are not made by running `make dist`.

I would therefore like to include the 8 files so that users can compile vsearch without having autoconf installed. They only need to download, unpack, run `./configure` and then `make`. No need to run `autogen.sh`.

**Comment by xflouris** [comment]
You're completely right. I was under the impression a manual upload of tarballs was possible for a release. All 8 files are then required.

**Comment by torognes** [comment]
Source tarballs are created automatically, but we can add other files, e.g. binary distributions.

**Comment by torognes** [comment]
Fixed in 1.9.0

**Comment by frederic-mahe** [comment]
Hi @torognes, hi @xflouris,

I just tried a fresh installation of vsearch on a new computer and I obtained the following error:
```sh
git clone https://github.com/torognes/vsearch.git
[...]
./configure
[...]
make
CDPATH="${ZSH_VERSION+.}:" && cd . && /bin/sh /homedir/mahe/Science/src/vsearch/missing aclocal-1.15 
/homedir/mahe/Science/src/vsearch/missing: line 81: aclocal-1.15: command not found
WARNING: 'aclocal-1.15' is missing on your system.
         You should only need it if you modified 'acinclude.m4' or
         'configure.ac' or m4 files included by 'configure.ac'.
         The 'aclocal' program is part of the GNU Automake package:
         <http://www.gnu.org/software/automake>
         It also requires GNU Autoconf, GNU m4 and Perl in order to run:
         <http://www.gnu.org/software/autoconf>
         <http://www.gnu.org/software/m4/>
         <http://www.perl.org/>
make: *** [aclocal.m4] Error 127
```
It seems that the ac files embedded in the GitHub repository are in conflict with the version of autotools installed on that new machine. If I launch `./autogen.sh` first, things are OK.

Additionally, the vsearch clone repository on my primary computer refused to pull new updates from GitHub: the new ac files embedded in the GitHub repository were in conflict with my own ac files. It forced me to discard local modifications (`git checkout -- configure aclocal.m4 Makefile.in man/Makefile.in src/Makefile.in`) to be able to `git pull`. On my computer too, the `make` command failed and I had to run `./autogen.sh` to be able to compile. 

I understand the idea of avoiding a dependency by embedding ac files, but it seems to be fragile. Most people will probably use the binaries. Maybe it is safe to assume that if someone wants to compile, that person will already have the autotools or will know how to install them.

My suggestion is to remove the ac files from the GitHub repository. What do you think?

**Comment by torognes** [comment]
I see your point @frederic-mahe. Perhaps we should drop those files and say that autotools (version 2.63) is required if you want to build from source. After all we do require other development tools like gcc etc in that case.

Alternatively, we could instruct those who already have autotools installed to run `autogen.sh`.

**Comment by frederic-mahe** [comment]
When you run `./autogen.sh`, you can end up with a different configuration file. When latter you try to pull a new version from github, you obtain this error:
```
git pull
remote: Counting objects: 27, done.
remote: Compressing objects: 100% (9/9), done.
remote: Total 27 (delta 14), reused 14 (delta 14), pack-reused 4
Unpacking objects: 100% (27/27), done.
From https://github.com/torognes/vsearch
   48e08db..4245338  master     -> origin/master
 * [new tag]         v1.10.2    -> v1.10.2
Updating 48e08db..4245338
error: Your local changes to 'configure' would be overwritten by merge.  Aborting.
Please, commit your changes or stash them before you can merge.
```
To be able to pull, I have to discard the modified file:
```
git checkout -- configure
git pull
```
In practice, I always run `./autogen.sh ; ./configure ; make -j`, so I am in favor of removing the files from the repository and indicate that autotools are required to compile vsearch.

**Comment by torognes** [comment]
Fixed in version 1.11.1 just released.
